src/SymmetricPositiveDefinite.jl

```diff
@@ -0,0 +1,176 @@
+using LinearAlgebra: svd, eig, eigen
+
+@doc doc"""
+    SymmetricPositiveDefinite{N} <: Manifold
+
+The manifold of symmetric positive definite matrices, i.e.
+
+```math
+\mathcal P(n) \coloneqq
+\bigl\{
+    x \in \mathbb R^{n\\times n} :
+    \xi^\mathrm{T}x\xi > 0 \text{ for all } \xi \in \mathbb R^{n}\backslash\{0\}
+\bigr\}
+```
+
+# Constructor
+
+    SymmetricPositiveDefinite(n)
+
+generates the $\mathcal P(n) \subset \mathbb R^{n\times n}$
+"""
+struct SymmetricPositiveDefinite{N} <: Manifold end
+SymmetricPositiveDefinite(n::Int) = SymmetricPositiveDefinite{n}()
+
+@doc doc"""
+    manifold_dimension(::SymmetricPositiveDefinite{N})
+
+returns the dimension of the manifold [`SymmetricPositiveDefinite`](@ref) $\mathcal P(n)$, N\in \mathbb N$, i.e.
+```math
+    \frac{n(n+1)}{2}    
+```
+"""
+@generated manifold_dimension(::SymmetricPositiveDefinite{N}) where {N} = N*(N+1)/2
+
+@doc doc"""
+
+"""
+struct LinearAffineMetric <: Metric end
+@traitimpl HasMetric{SymmetricPositiveDefinite, LogEuclidean}
+
+@doc doc"""
+
+"""
+struct LogEuclideanMetric <: Metric end
+@traitimpl HasMetric{SymmetricPositiveDefinite, LogEuclidean}
+
+distance(P::SymmetricPositiveDefinite{N},x,y) = distance(MetricManifold(S,LinearAffineMetric)},x,y)
```

Comment:
I think this is a more natural approach for our manifolds than what we've done so far. I think it makes sense for us to implement our manifold functions with explicit metrics (e.g. the line below) and then forward the functions for the manifold itself to those `MetricManifold` functions for clarity, instead of using the `HasMetric` approach. I envision `HasMetric` for the user who wants to just make their manifold and get the `MetricManifold` for free, but it would probably be clearer if we proceeded as you've done here. Likewise, if an operation on a manifold is more efficient if a group structure is assumed, it might make sense in some cases to forward those operations to the corresponding group for clarity.

Reply:
Also, it should be `distance(MetricManifold(P,LinearAffineMetric())},x,y)`

Reply:
And, as explained shortly above, I thought the `Trait` merely indicates `A has Metric B` as if: Some functions check that. But actually while implementing I liked this fallback idea. But one could even mix that I think: For a `hasMetric`(that should then be `hasDefaultMetric`-Trait one could globally define the fallback, i.e. that `f(a,...) = f(MetricManifold{A,B},...)` if that trait exists (if such a thing is possible, I don't know.).

Reply:
Yeah, that isn't possible with multi-parameter traits, which is frustrating, because you're right, it is more intuitive. `HasDefaultMetric` does make more sense as a name than `HasMetric`.